stream_alert/threat_intel_downloader/threat_stream.py

```diff
@@ -0,0 +1,260 @@
+                - Third object is bool to indicated if retrieve more IOCs from
+                    threat feed.
+        """
+        if not event:
+            return None, None, False
+
+        next_url = None
+        if 'next_url' not in event:
+            next_url = '/api/v2/{}/?username={}&api_key={}&status={}&limit={}'\
+                .format(self._API_RESOURCE,
+                        self.api_user,
+                        self.api_key,
+                        self._IOC_STATUS,
+                        self._API_MAX_LIMIT)
+        else:
+            next_url = event.get('next_url', None)
+
+        if not next_url:
+            return None, None, False
+
+        return self._connect(next_url)
+
+    def _process_data(self, data):
+        """Process and filter data by sources and keys
+        Args:
+            data (list): A list contains ioc information
+                Example:
+                    [
+                        {
+                            'value': 'malicious_domain.com',
+                            'itype': 'c2_domain',
+                            'source': 'crowdstrike',
+                            'type': 'domain',
+                            'expiration_ts': '2017-12-19T04:45:18.412Z',
+                            'key1': 'value1',
+                            'key2': 'value2',
+                            ...
+                        },
+                        {
+                            'value': 'malicious_domain2.com',
+                            'itype': 'c2_domain',
+                            'source': 'ioc_source2',
+                            'type': 'domain',
+                            'expiration_ts': '2017-12-31T04:45:18.412Z',
+                            'key1': 'value1',
+                            'key2': 'value2',
+                            ...
+                        }
+                    ]
+
+        Returns:
+            (list): A list of dict contains useful IOC information
+                Example:
+                    [
+                        {
+                            'value': 'malicious_domain.com',
+                            'itype': 'c2_domain',
+                            'source': 'crowdstrike',
+                            'type': 'domain',
+                            'expiration_ts': 1513658718,
+                        }
+                    ]
+        """
+        def _epoch_time(time_str):
+            """Convert expiration time (in UTC) to epoch time
+            Args:
+                time_str (str): expiration time in string format
+                    Example: '2017-12-19T04:45:18.412Z'
+
+            Returns:
+                (int): Epoch time. If no expiration time presented, return to
+                    default value which is current time + 90 days.
+            """
+
+            if not time_str:
+                return int((datetime.now() + timedelta(90) - datetime(1970, 1, 1)).total_seconds())
```

Comment:
I think the preferred way for the latter part of this (`datetime(1970, 1, 1)`) would be `datetime.utcfromtimestamp(0)`.

Can you also add the `days=90` keyword arg to the `timedelta(90)` call, since timedelta can also take `seconds=`, `minutes=` - it just helps with readability without having to reference the docstring above.